app/src/main/java/space/taran/arknavigator/mvp/model/repo/kind/ResourceKind.kt

```diff
@@ -0,0 +1,79 @@
+package space.taran.arknavigator.mvp.model.repo.kind
+
+import space.taran.arknavigator.mvp.model.dao.ResourceExtra
+import space.taran.arknavigator.utils.extension
+import java.nio.file.Path
+import kotlin.reflect.full.isSubclassOf
+
+sealed class ResourceKind {
+    val kindCode = sealedSubClasses.indexOf(this::class)
+
+    object Image : ResourceKind()
+
+    class Video(
+        val height: Long? = null,
+        val width: Long? = null,
+        val duration: Long? = null
+    ) : ResourceKind()
+
+    class Document(val pages: Int? = null) : ResourceKind()
+
+    class Link(
+        val title: String? = null,
+        val description: String? = null,
+        val url: String? = null
+    ) : ResourceKind()
+
+    companion object {
+        private val sealedSubClasses = ResourceKind::class.sealedSubclasses.filter {
```

Comment:
Is it possible only with _sealed_ subclasses?

I would get rid of reflection at all and use enums again :) Reflection in general is slower and unreliable. Although we don't need speed when we extract subclasses, what I don't like in this file is that for every new kind developer would need to put declaration of it here. I would just make field-list containing all necessary implementations, and put everything related to particular kinds into separate files.

Also if it is possible to hide stuff like
```
Video::class ->
                    Video(
                        data[MetaExtraTag.HEIGHT]?.toLong(),
                        data[MetaExtraTag.WIDTH]?.toLong(),
                        data[MetaExtraTag.DURATION]?.toLong()
                    )
```
into kind-specific file, it would be also good.

Otherwise we risk overbloat this file when we add 10 or more kinds. Also, adding new kind, ideally, should be copying existing file and modifying it.

Refactoring looks nice in the rest of places.